### PR TITLE
Improved sanitizer support

### PR DESCRIPTION
**Pull request by DmitriyMusatkin**
*Issue #, if available:*

*Description of changes:*
Fixed sanitizer cmake to work cross-platform and annotations to also work cross platform.
Removed sanitizer blacklist that only works on clang

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cmake/AwsSanitizers.cmake [hunk 1]
```diff
@@ -39,40 +39,39 @@ endfunction()
 # This function enables sanitizers on the given target
 # Options:
 #  SANITIZERS: The list of extra sanitizers to enable
-#  BLACKLIST: The blacklist file to use (passed to -fsanitizer-blacklist=)
 function(aws_add_sanitizers target)
     set(oneValueArgs BLACKLIST)
```
Comment (waahm7) [inline cmake/AwsSanitizers.cmake:43]: nit: we can remove this.

**Review by waahm7 — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -121,7 +122,8 @@ jobs:
     - name: Build ${{ env.PACKAGE_NAME }} + consumers
       run: |
         python -c "from urllib.request import urlretrieve; urlretrieve('${{ env.BUILDER_HOST }}/${{ env.BUILDER_SOURCE }}/${{ env.BUILDER_VERSION }}/builder.pyz?run=${{ env.RUN }}', 'builder.pyz')"
-        python builder.pyz build -p ${{ env.PACKAGE_NAME }} --target windows-${{ matrix.arch }} --compiler msvc-16
+        python builder.pyz build -p ${{ env.PACKAGE_NAME }} --target windows-${{
+        matrix.arch }} --compiler msvc-16
```
Comment (graebm) [inline .github/workflows/ci.yml:126]: trivial: Weird change. Was this on purpose? seems like a weird change to make when the line before this is way longer

if you want to split this across multiple lines, can we do it the traditional way with backslashes? 
```
python builder.pyz build -p ${{ env.PACKAGE_NAME }} \
  --target windows-${{ matrix.arch }} --compiler msvc-16
```

**Review by graebm — APPROVED** [review]
nice work!

File: .github/workflows/ci.yml (hunk 2 above)
Comment (DmitriyMusatkin) [reply .github/workflows/ci.yml:126]: ah, i was experimenting with some flags on the ci runs and i think my ide decided to split it into 2 lines. let me revert it

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -143,7 +144,8 @@ jobs:
     - name: Build ${{ env.PACKAGE_NAME }} + consumers
       run: |
         python -c "from urllib.request import urlretrieve; urlretrieve('${{ env.BUILDER_HOST }}/${{ env.BUILDER_SOURCE }}/${{ env.BUILDER_VERSION }}/builder.pyz?run=${{ env.RUN }}', 'builder.pyz')"
-        python builder.pyz build -p ${{ env.PACKAGE_NAME }} --target windows-${{ matrix.arch }} --compiler msvc-14
+        python builder.pyz build -p ${{ env.PACKAGE_NAME }} --target windows-${{
+        matrix.arch }} --compiler msvc-14
```
Comment (graebm) [inline .github/workflows/ci.yml:148]: one more mystery line break
```suggestion
        python builder.pyz build -p ${{ env.PACKAGE_NAME }} --target windows-${{ matrix.arch }} --compiler msvc-14
```

**Pull request merged**